src/gap-skipper.js

```diff
@@ -0,0 +1,176 @@
+    // Set of events that reset the gap-skipper logic and clear the timeout
+    let timerCancelEvents = [
+      'seeking',
+      'seeked',
+      'pause',
+      'playing',
+      'error'
+    ];
+
+    let cancelTimerHandler = this.cancelTimer_.bind(this);
+
+    this.tech_.on('waiting', waitingHandler);
+    this.tech_.on('timeupdate', timeupdateHandler);
+    this.tech_.on(timerCancelEvents, cancelTimerHandler);
+
+    this.dispose = () => {
+      this.logger_('dispose');
+      this.tech_.off('waiting', waitingHandler);
+      this.tech_.off('timeupdate', timeupdateHandler);
+      this.tech_.off(timerCancelEvents, cancelTimerHandler);
+      this.cancelTimer_();
+    };
+  }
+
+  /**
+  * Cancels any pending timers and resets the 'timeupdate' mechanism
+  * designed to detect that we are stalled
+  *
+  * @private
+  */
+  cancelTimer_() {
+    this.consecutiveUpdates = 0;
+
+    if (this.timer_) {
+      this.logger_('cancelTimer_');
+      clearTimeout(this.timer_);
+    }
+
+    this.timer_ = null;
+  }
+
+  /**
+   * Timer callback. If playback still has not proceeded, then we seek
+   * to the start of the next buffered region.
+   *
+   * @private
+   */
+  skipTheGap_(scheduledCurrentTime) {
+    let buffered = this.tech_.buffered();
+    let currentTime = this.tech_.currentTime();
+    let nextRange = Ranges.findNextRange(buffered, currentTime);
+
+    this.consecutiveUpdates = 0;
+    this.timer_ = null;
+
+    if (nextRange.length === 0 ||
+        currentTime !== scheduledCurrentTime) {
+      return;
+    }
+
+    this.logger_('skipTheGap_:',
+                 'currentTime:', currentTime,
+                 'scheduled currentTime:', scheduledCurrentTime,
+                 'nextRange start:', nextRange.start(0));
+
+    // only seek if we still have not played
+    this.tech_.setCurrentTime(nextRange.start(0) + Ranges.TIME_FUDGE_FACTOR);
+  }
+
+  /**
+   * Set a timer to skip the unbuffered region.
+   *
+   * @private
+   */
+  setTimer_() {
+    let buffered = this.tech_.buffered();
+    let currentTime = this.tech_.currentTime();
+    let nextRange = Ranges.findNextRange(buffered, currentTime);
+
+    if (nextRange.length === 0 ||
+        this.timer_ !== null) {
+      return;
+    }
+
+    let difference = nextRange.start(0) - currentTime;
+
+    this.logger_('setTimer_:',
+                 'stopped at:', currentTime,
+                 'setting timer for:', difference,
+                 'seeking to:', nextRange.start(0));
+
+    this.timer_ = setTimeout(this.skipTheGap_.bind(this),
+                             difference * 1000,
+                             currentTime);
+  }
+
+  /**
+   * A debugging logger noop that is set to console.log only if debugging
+   * is enabled globally
+   *
+   * @private
+   */
+  logger_() {}
+
+  /**
+   * A noop to ensure there is always have a dispose function even if there
+   * was no playerId in the global options and therefore the gapSkipper was
+   * never properly initialized
```

Comment:
This sounds like `playerId` is a required parameter. You should throw an error instead of returning a broken object in this case.